core/src/avm2/activation.rs

```diff
@@ -1203,22 +1303,23 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     ) -> Result<FrameControl<'gc>, Error> {
         let multiname = self.pool_multiname(method, index)?;
         let object = self.context.avm2.pop().coerce_to_object(self)?;
-        let base_proto: Result<Object<'gc>, Error> = self
-            .base_proto()
-            .and_then(|p| p.proto())
-            .ok_or_else(|| "Attempted to get property on non-existent super object".into());
-        let base_proto = base_proto?;
-        let mut base = base_proto.construct(self, &[])?; //TODO: very hacky workaround
-
-        let name: Result<QName, Error> = base.resolve_multiname(&multiname)?.ok_or_else(|| {
-            format!(
-                "Could not resolve {:?} as super property",
-                multiname.local_name()
-            )
-            .into()
-        });
 
-        let value = base.get_property(object, &name?, self)?;
+        let name: Result<QName, Error> = object
+            .resolve_multiname(&multiname)?
+            .ok_or_else(|| format!("Could not find method {:?}", multiname.local_name()).into());
+        let name = name?;
+
+        let superclass_object: Result<Object<'gc>, Error> = self
+            .subclass_object()
+            .and_then(|bc| bc.superclass_object())
+            .ok_or_else(|| {
+                "Attempted to call super method without a superclass."
+                    .to_string()
+                    .into()
+            });
+        let superclass_object = superclass_object?;
+
+        let value = superclass_object.call_instance_getter(&name, Some(object), self)?;
```

Comment:
This is still incorrectly requiring that the property is a getter, and `setsuper` requiring that the property is a setter. Try:

```
class Foo {
	public var a:int = 2;
}

class Bar extends Foo {
	function f() {
		trace(a);
		trace(super.a);
		super.a = 10;
		trace(a);
		trace(super.a);
	}
}

var bar:Bar = new Bar();
bar.f();
```